### ISSUE TITLE
How to improve detection o small objects, football balls

### ISSUE THREAD

**Issue by keko950** [AlexeyAB/darknet#3339]

**Comment by AlexeyAB** [comment]
@keko950 Hi,

* What model do you use, is it `yolov3-spp.cfg`?
* What mAP do you get on Training and Detection datasets?
* What is your image sizes?

**Comment by keko950** [comment]
@AlexeyAB Hey, yes i use yolov3-spp.
In balls i get near 73%, players 93%.
Image size 512x512, no resizing
This is an example detection image:
![2](https://user-images.githubusercontent.com/9590497/59191320-7b76c700-8b7f-11e9-8a70-185c929cba60.png)

![1](https://user-images.githubusercontent.com/9590497/59191294-6bf77e00-8b7f-11e9-8df3-78a1f2d501a6.png)

**Comment by AlexeyAB** [comment]
> How to improve detection o small objects, football balls

Just collect more images to training dataset.
And train the model with `width=832 height=832` in cfg-file

**Comment by keko950** [comment]
Yeah, i think i need more dataset. Currently i have 30000 training frames, per 3000 validation frames.
Loss is about 1.5 and doesnt go lower so.. 
But why would i want to train with 832x832? I mean, im not resizing images, i will get the same quality.

**Comment by AlexeyAB** [comment]
Oh, yes, if all your images 512x512 then use width=512 height=512

If you will use only this repository, then you can try to train by using new models: https://github.com/AlexeyAB/darknet/issues/3114
For example: https://github.com/AlexeyAB/darknet/files/3253820/yolo_v3_spp_pan_scale.cfg.txt

**Comment by keko950** [comment]
Yeah, im using only this repo. I will try them, thanks!

**Comment by keko950** [comment]
One more question @AlexeyAB , which weights should i use to train panet? maybe darknet53.conv.74?

**Comment by AlexeyAB** [comment]
Yes

**Comment by keko950** [comment]
Hi again @AlexeyAB , just wanted to tell you i've got very good results from this last night. I've changed yolov3-spp.cfg, learning rate to = 0.005,  saturation = 2, exposure=2  and seems to improve way better. From 73% to 83%, and loss decreased a bit (from 1.5 to 1.2), should now i decrease learning rate again? or perhaps should i keep training with the 0.005?

![chart](https://user-images.githubusercontent.com/9590497/59257437-3eb9d700-8c36-11e9-9912-22340f120e36.png)

**Comment by keko950** [comment]
@AlexeyAB Currently testing the last version of the repo with the pan scale model, but im receiving this error sometimes while the training:

 " Wrong annotation: x = 0, y = 0, < 0 or > 1 

 Wrong annotation: x = 0, y = 0, < 0 or > 1 " 


did the annotation method changed at some point?

**Comment by keko950** [comment]
> @AlexeyAB Currently testing the last version of the repo with the pan scale model, but im receiving this error sometimes while the training:
> 
> " Wrong annotation: x = 0, y = 0, < 0 or > 1
> 
> Wrong annotation: x = 0, y = 0, < 0 or > 1 "
> 
> did the annotation method changed at some point?

NVM, problem with my dataset

**Comment by AlexeyAB** [comment]
@keko950 Check your dataset, and look in files bad.lisl, bad_label.list )
